### PR TITLE
[GitHub Notifier] Add deployment configs.

### PR DESCRIPTION
**Pull request by wlynch**
# Changes

This adds deployment and related configs for turning up the notifier in
the dogfooding environment. Namespace, secret name, etc are taken from
the dogfooding environment.

Also bumps knative.dev/pkg dependency, since the old version was masking
permission denied errors when starting up the controller.

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [n/a] Includes [docs](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if user facing)
- [x] Commit messages follow [commit message best practices](https://github.com/tektoncd/community/blob/master/standards.md#commit-messages)
- [x] Commit messages includes a project tag in the subject - e.g. [OCI], [hub], [results], [task-loops]

_See [the contribution guide](https://github.com/tektoncd/experimental/blob/master/CONTRIBUTING.md)
for more details._

### CODE REVIEW

File: notifiers/github-app/cmd/main.go [hunk 1]
```diff
@@ -60,9 +61,14 @@ func main() {
 		log.Fatalf("error creating github client: %v", err)
 	}
 
-	config, err := clientcmd.BuildConfigFromFlags("", filepath.Join(os.Getenv("HOME"), ".kube", "config"))
+	// creates the in-cluster config
+	config, err := rest.InClusterConfig()
 	if err != nil {
-		log.Fatalf("error getting kubernetes client config: %v", err)
+		log.Println("unable to find in-cluster kubeconfig. Falling back to local config")
+		config, err = clientcmd.BuildConfigFromFlags("", filepath.Join(os.Getenv("HOME"), ".kube", "config"))
```
Comment (dibyom) [inline notifiers/github-app/cmd/main.go:68]: I think we can just use `BuildConfigFromFlags` . If no flags are specified, it defaults to rest.InClusterConfig

https://github.com/kubernetes/client-go/blob/v0.21.1/tools/clientcmd/client_config.go#L614-L616

**Review by dibyom — APPROVED** [review]
One nit, otherwise looks good!

File: notifiers/github-app/cmd/main.go (hunk 1 above)
Comment (wlynch) [reply notifiers/github-app/cmd/main.go:68]: Done (though used knative libraries to keep to avoid flag conflicts).

**Pull request merged**